conf/nginx-testing.conf

```diff
@@ -57,40 +57,28 @@ server {
-               add_header 'Access-Control-Expose-Headers' 'Content-Length,Content-Range';
+               add_header 'Access-Control-Allow-Headers' 'DNT, User-Agent, X-Requested-With, If-Modified-Since, Cache-Control, Content-Type, Range, X-Auth, Accept, Origin';
             }
 
             if ($request_method = 'PUT') {
-               add_header 'Access-Control-Allow-Origin' '*';
-               add_header 'Access-Control-Allow-Methods' 'GET, POST, PUT, DELETE, PATCH, OPTIONS';
-               add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range, access-control-allow-origin';
-               add_header 'Access-Control-Expose-Headers' 'Content-Length,Content-Range';
+               add_header 'Access-Control-Allow-Headers' 'DNT, User-Agent, X-Requested-With, If-Modified-Since, Cache-Control, Content-Type, Range, X-Auth, Accept, Origin';
             }
 
             if ($request_method = 'POST') {
-               add_header 'Access-Control-Allow-Origin' '*';
-               add_header 'Access-Control-Allow-Methods' 'GET, POST, PUT, DELETE, PATCH, OPTIONS';
-               add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range, access-control-allow-origin';
-               add_header 'Access-Control-Expose-Headers' 'Content-Length,Content-Range';
+               add_header 'Access-Control-Allow-Headers' 'DNT, User-Agent, X-Requested-With, If-Modified-Since, Cache-Control, Content-Type, Range, X-Auth, Accept, Origin';
             }
 
 
             if ($request_method = 'DELETE') {
-               add_header 'Access-Control-Allow-Origin' '*';
-               add_header 'Access-Control-Allow-Methods' 'GET, POST, PUT, DELETE, PATCH, OPTIONS';
-               add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range, access-control-allow-origin';
-               add_header 'Access-Control-Expose-Headers' 'Content-Length,Content-Range';
+               add_header 'Access-Control-Allow-Headers' 'DNT, User-Agent, X-Requested-With, If-Modified-Since, Cache-Control, Content-Type, Range, X-Auth, Accept, Origin';
             }
 
             if ($request_method = 'OPTIONS') {
-               add_header 'Access-Control-Allow-Origin' '*';
                add_header 'Access-Control-Allow-Methods' 'GET, POST, PUT, DELETE, PATCH, OPTIONS';
-               add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range, Access-Control-Allow-Origin';
-               add_header 'Access-Control-Expose-Headers' 'Content-Length,Content-Range';
+               add_header 'Access-Control-Allow-Headers' 'DNT, User-Agent, X-Requested-With, If-Modified-Since, Cache-Control, Content-Type, Range, X-Auth, Accept, Origin';
```

Comment:
didn t we agree to merge these up front and leave all the if conditions? or is it actually needed now?

Reply:
In the end, it didn't work as expected when I left the conditions completely.
I tried to reduce it as much as possible.

Reply:
could you please add a some comment there why it is needed so somebody will not just remove it for same reason we wanted to? It would make sense to any committer and we would create a bug that way.